manager/controlapi/service.go

```diff
@@ -238,6 +286,10 @@ func (s *Server) UpdateService(ctx context.Context, request *api.UpdateServiceRe
 		return nil, err
 	}
 
+	if err := s.checkPortConflicts(request.Spec); err != nil {
+		return nil, err
+	}
```

Comment:
I think it would be a worthwhile optimization to fetch the existing service first and see if `spec.Endpoint` changed at all relative to the previous version (`reflect.DeepEqual` should be sufficient). We can skip the expensive `checkPortConflicts` call if there's no change to the endpoint settings.